service/history/workflow/context.go

```diff
@@ -234,8 +234,12 @@ func (c *ContextImpl) Clear() {
 	metrics.WorkflowContextCleared.With(c.metricsHandler).Record(1)
 	if c.MutableState != nil {
 		c.MutableState.GetQueryRegistry().Clear()
+		c.MutableState = nil
+	}
+	if c.updateRegistry != nil {
+		c.updateRegistry.Clear()
+		c.updateRegistry = nil
```

Comment:
In order to obtain the benefits of this PR, do you need to discard the Admitted updates, or would an alternative be to clear only the Accepted and Completed updates from the registry, and allow them to be reloaded from MutableState when we next access the registry?